spec/ruby/core/method/parameters_spec.rb

```diff
@@ -22,6 +22,8 @@ def one_splat_one_block(*args, &block)
       local_is_not_parameter = {}
     end
 
+    def one_noblock(&nil); end
```

Comment:
I think this needs to be guarded in a `ruby_version_is '3.4'` block.  Same for the two other spec changes (no test changes needed, as tests always assume the master branch is used).

Reply:
@eregon I had to use `eval` for this, even after wrapping the new methods inside a `ruby_version_is` block, since the new syntax is (understandably) a syntax error on old Rubies.

Is this the right way to do this, or am I missing something? I also couldn't find any guidance about syntax changes in the README or CONTRIBUTING docs inside the `spec` folder.

Reply:
Yes, that's fine and already used in some other places, notably language/ specs.